### PR TITLE
Use readline only when stdin is a tty

### PR DESCRIPTION
**Pull request by pepone**
With this PR readline is only used when stdin is a tty

### CODE REVIEW

File: cpp/src/IceGrid/Parser.cpp [hunk 1]
```diff
@@ -2573,70 +2573,77 @@ Parser::getInput(char* buf, size_t& result, size_t maxSize)
...
+                {
+                    add_history(line);
+                }
+
+                result = strlen(line) + 1;
+                if(result > maxSize)
+                {
+                    free(line);
+                    error("input line too long");
+                    result = 0;
```
Comment (bernardnormier) [inline cpp/src/IceGrid/Parser.cpp:2596]: Should be 1

**Review by bernardnormier — APPROVED** [review]
Looks good.

File: cpp/src/IceGrid/Parser.cpp (hunk 1 above)
Comment (pepone) [reply cpp/src/IceGrid/Parser.cpp:2596]: Why 1? I think this is just the number of bytes read and we didn't read anything.

Comment (bernardnormier) [reply cpp/src/IceGrid/Parser.cpp:2596]: My mistake, I thought it was an exit status. 0 is fine.

**Comment by mreinart** [comment]
Hi - only FYI
latest change induced need to install libedit-dev on linux (rpi) - did not find <editline/readline.h>
KR Manfred

**Comment by externl** [comment]
Hi @mreinart, not sure what you mean. Are you saying you installed `libdedit-dev` but still could not find `<editline/readline.h>`, or maybe that you needed to install `libedit-dev` to get `<editline/readline.h>`?

**Comment by mreinart** [comment]
Hi @externl,
I only intended to inform about the latter case - `libdedit-dev` was additionally required for successful compilation/build.
Naively I had tried similarly named `libeditline-dev` first - which did not help as it contains some different `editline.h` - thus I considered worth informing to avoid confusion for others.

**Pull request merged**